File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-2.kt
class Solution {
    fun removeDuplicates(nums: IntArray): Int {
        var res = 0

            for (i in nums.indices) {
                if (i < nums.size - 1 && nums[i] == nums[i + 1]) {
                    continue
                } else {
                    nums[res] = nums[i]
                    res++
                }
            }

            return res
    }
}

File: Data Structures & Algorithms/maximum-product-subarray/submission-2.py
class Solution:
    def maxProduct(self, nums: List[int]) -> int:
        res = max(nums) 
        currMin, currMax = 1, 1
        for n in nums:
            tmp = n * currMax
            currMax = max(n*currMax, n * currMin, n)
            currMin = min(n*currMin, tmp, n)
            res = max(currMax, res)
        return res